### PR TITLE
Trip model fix

### PR DESCRIPTION
**Pull request by mimizotti**
I made changes to the `trip` model and fixed two bugs.

### CODE REVIEW

File: app/models/trip.rb [hunk 1]
```diff
@@ -116,7 +116,7 @@ def self.most_frequent_origination(station_id)
   end
 
   def self.date_with_highest_trip_from_this_station(station_id)
-    date = Trip.where(start_station_id: 1).group('(EXTRACT(MONTH FROM start_date))::integer').group('(EXTRACT(DAY FROM start_date))::integer').group('(EXTRACT(YEAR FROM start_date))::integer').order('count_all').reverse_order.count.first
+    date = Trip.where(start_station_id: station_id).group('(EXTRACT(MONTH FROM start_date))::integer').group('(EXTRACT(DAY FROM start_date))::integer').group('(EXTRACT(YEAR FROM start_date))::integer').order('count_all').reverse_order.count.first
```
Comment (snayrouz) [inline app/models/trip.rb]: Looks good. Great catch with the `station_id`

File: app/models/trip.rb [hunk 2]
```diff
@@ -96,37 +96,65 @@ def self.count_of_fewest_trips
   end
 
   def self.number_of_starting_rides_at_station(station_id)
-    stations = Trip.group(:start_station_id).count(:id)
-    stations[station_id]
+    if Trip.where(start_station_id: station_id) == []
```
Comment (amhursh) [inline app/models/trip.rb:99]: I think this is good/best practice for us on things like this in the future, even if it wasn't really our problem, after all. Way to go!

Comment (mimizotti) [reply app/models/trip.rb:99]: Thanks, Aaron!

Comment (snayrouz) [inline app/models/trip.rb:99]: Good idea and implementation of what needs to happen when our queries return nothing

File: app/models/trip.rb [hunk 3]
```diff
@@ -96,37 +96,65 @@ def self.count_of_fewest_trips
...
+    else
+      station = Trip.where(start_station_id: station_id).group(:end_station_id).order('count_id DESC').count(:id).first
+      Station.find(station[0])[:name]
+    end
   end
 
   def self.most_frequent_origination(station_id)
-    station = Trip.where(end_station_id: station_id).group(:start_station_id).order('count_id DESC').count(:id).first
-    Station.find(station[0])[:name]
+    if Trip.where(start_station_id: station_id) == []
```
Comment (snayrouz) [inline app/models/trip.rb:126]: @mimilettd Oh, I see you did this for all this methods! Awesome, thanks Mimi!

File: app/views/trips/dashboard.erb [hunk 4]
```diff
@@ -4,12 +4,12 @@
...
   <p>Shortest ride: <%= Trip.shortest_ride %></p>
-  <p>Station with the most rides as a starting place: <%= (Trip.station_with_most_ride_starts).name %></p>
-  <p>Station with the most rides as an ending place: <%= (Trip.station_with_most_rides_ends).name %></p>
+  <p>Station with the most rides as a starting place: <%= (Trip.station_with_most_ride_starts) %></p>
+  <p>Station with the most rides as an ending place: <%= (Trip.station_with_most_rides_ends) %></p>
   <p>Month by Month breakdown of number of rides with subtotals for each year: <%= Trip.month_breakdown %></p>
   <p>Most ridden bike with total number of rides for that bike: <%= Trip.most_ridden_bike %></p>
   <p>Least ridden bike with total number of rides for that bike: <%= Trip.least_ridden_bike %></p>
-  <p>User subscription type breakout with both count and percentage: <%= Trip.subscription_type %></p>
+  <p>User subscription type breakout with both count and percentage: <%= Trip.subscription_info %></p>
```
Comment (amhursh) [inline app/views/trips/dashboard.erb:12]: Thanks for fixing this bug!

File: app/views/trips/dashboard.erb [hunk 5]
```diff
@@ -4,12 +4,12 @@
...
   <p>Longest ride: <%= Trip.longest_ride %></p>
   <p>Shortest ride: <%= Trip.shortest_ride %></p>
-  <p>Station with the most rides as a starting place: <%= (Trip.station_with_most_ride_starts).name %></p>
-  <p>Station with the most rides as an ending place: <%= (Trip.station_with_most_rides_ends).name %></p>
+  <p>Station with the most rides as a starting place: <%= (Trip.station_with_most_ride_starts) %></p>
+  <p>Station with the most rides as an ending place: <%= (Trip.station_with_most_rides_ends) %></p>
   <p>Month by Month breakdown of number of rides with subtotals for each year: <%= Trip.month_breakdown %></p>
   <p>Most ridden bike with total number of rides for that bike: <%= Trip.most_ridden_bike %></p>
   <p>Least ridden bike with total number of rides for that bike: <%= Trip.least_ridden_bike %></p>
-  <p>User subscription type breakout with both count and percentage: <%= Trip.subscription_type %></p>
```
Comment (snayrouz) [inline app/views/trips/dashboard.erb:12]: Good catch!

File: app/views/trips/index.erb [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <h1>All Trips</h1>
 
 <div>
-  <% @trips.each do |trip| %>
+  <% @trips[1..30].each do |trip| %>
```
Comment (snayrouz) [inline app/views/trips/index.erb:4]: This is great, we just have to remember to take it out.

Comment (amhursh) [inline app/views/trips/index.erb:4]: We will have to switch this out with paginator magic.

File: db/migrate/20170822181831_change_conditions.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      t.remove :min_sea_level_pressure_inches
+      t.remove :max_visibility_miles
+      t.remove :min_visibility_miles
+      t.remove :max_wind_Speed_mph
+      t.remove :max_gust_speed_mph
+      t.remove :cloud_cover
+      t.remove :events
+      t.remove :wind_dir_degrees
+  end
+
```
Comment (snayrouz) [inline db/migrate/20170822181831_change_conditions.rb:20]: I'm assuming this is all just from dropping and migrating. But cool, looks good.

**Review by snayrouz — COMMENTED** [review]
These look good, ready to merge on my end. Will wait for @amhursh final decision

File: app/models/trip.rb (hunk 1 above)
Comment (amhursh) [reply app/models/trip.rb]: Yeah this was a fun bug. Good catch.

**Pull request merged**